### ISSUE TITLE
MethodError for `round` on `Dates.Time` types

### ISSUE THREAD

**Issue by adknudson** [JuliaLang/julia#56394]
The method `round` is defined for `Dates.TimeTypes`, but fails when the object is a `Dates.Time` type due to `floor` not being defined for `Dates.Time`.

```julia
julia> using Dates

julia> dt = now(); t = Time(dt);

julia> round(dt, Second(30)) # rounding DateTime works
2024-10-30T13:35:30

julia> round(t, Second(30))
ERROR: MethodError: no method matching floor(::Time, ::Second)
The function `floor` exists, but no method is defined for this combination of argument types.
```

Is there a motivating reason why `floor` is not defined for `Time` types?

**Comment by omus** [comment]
The `Time` type was added after the date rounding methods were added and I think this was just an oversight

**Comment by inkydragon** [comment]
Closed as a duplicate of #43284
